webrecorder/browsertrix
# feat: Public org profile page

Resolves #2156
Resolves #2133
Follows https://github.com/webrecorder/browsertrix/pull/2174 (must be deployed to dev for manual testing steps)

## Changes

- Enables creating a public org profile page with description and website at `/profile/<org slug>`
- Updates current "Overview" page to be "Dashboard", found under `/dashboard`
- Organizes org "General" settings tab by "General", "Profile", and "Developer Tools"
- Adds sign up banner to log in page for consistent CTA banners
- Updates copy and docs to support changes

## Manual testing

1. Log in as crawler
2. Go to org "Settings"
3. Update profile description and website. Verify values save as expected
4. Click "Preview public profile page". Verify banner is shown with text describing that profile is still private
5. Log out
6. Go to `/profile/<your org slug>`. Verify generic "not found" message is shown
7. Log back in
8. Go back to org settings
9. Enable "Allow anyone to see org"
10. Go back to public profile page and refresh. Verify preview banner is no longer shown, while controls to edit the page are still visible
11. Log out
12. Go to `/profile/<your org slug>`. Verify page is still visible

## Screenshots

### Org Settings
<img width="1028" alt="Screenshot 2024-11-26 at 2 35 23 PM" src="https://github.com/user-attachments/assets/3ac0369f-eac4-409e-82c7-05a75a5d1b61">

<img width="1030" alt="Screenshot 2024-11-26 at 2 35 46 PM" src="https://github.com/user-attachments/assets/779d7a59-827f-4d5f-af5f-e80973dc1f0d">

<img width="1032" alt="Screenshot 2024-11-26 at 2 35 52 PM" src="https://github.com/user-attachments/assets/ad85a9b3-c1df-47e5-9e3b-654cb241288e">

### Org Profile (private preview)

<img width="794" alt="Screenshot 2024-11-26 at 2 57 36 PM" src="https://github.com/user-attachments/assets/e8cbe0dd-f0f4-4abd-bff6-1741c9dc42c7">


### Org Profile (public)
<img width="802" alt="Screenshot 2024-11-26 at 2 57 54 PM" src="https://github.com/user-attachments/assets/d293bc48-5ab1-43f8-9f92-9a0fef485a54">



### Sign In
<img width="802" alt="Screenshot 2024-11-26 at 2 56 43 PM" src="https://github.com/user-attachments/assets/7228dcce-176a-424b-a4df-2499d9ba7cd7">




### Generic 404

<img width="797" alt="Screenshot 2024-11-26 at 2 56 54 PM" src="https://github.com/user-attachments/assets/b40245b9-4cd7-45b5-9129-fd79020ae248">


### Docs: User Guide > Org Settings > Change Org Settings
<img width="729" alt="Screenshot 2024-11-26 at 3 29 39 PM" src="https://github.com/user-attachments/assets/5beb2e9e-4dbe-4010-a37f-1114b00154a6">

## Follow-ups

This change doesn't include actual public collections yet, which will be addressed with https://github.com/webrecorder/browsertrix/issues/2182
